### ISSUE TITLE
Backslashes on $schema property (Windows)

### ISSUE THREAD

**Issue by gnuechtel** [nrwl/nx#10303]
## Current Behavior

When a new plugin project is created or existing plugin projects are migrated, the `$schema` property contains backslahes in paths.

I am not sure, if this affected by plugin projects only or it is a general issue.

It is a similar problem like #10122 

Wouldn't it be a good idea to add some general tests to ensure that all generators on Windows never use backslashes?
I love Nx and like to write tests. So if I could contribute in a feasible way, I would be exited.

## Expected Behavior

The `$schema` property should use forward slashes.

## Steps to Reproduce

```
npx create-nx-workspace@14.1.5 another-monorepo --preset=apps
cd another-monorepo
npm install @nrwl/nx-plugin
npx nx generate @nrwl/nx-plugin:plugin another-plugin
```

The files `libs/another-plugin/project.json` and `apps/another-plugin-e2e/project.json` contain:
```
"$schema": "..\\..\\node_modules\\nx\\schemas\\project-schema.json"
```

### Environment

```
   Node : 16.13.2
   OS   : win32 x64
   npm  : 8.1.2

   nx : 14.1.5
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.1.5
   @nrwl/eslint-plugin-nx : 14.1.5
   @nrwl/express : Not Found
   @nrwl/jest : 14.1.5
   @nrwl/js : 14.1.5
   @nrwl/linter : 14.1.5
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : 14.0.3
   @nrwl/nx-plugin : 14.1.5
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/workspace : 14.1.5
   typescript : 4.6.4
   rxjs : 6.6.7
   ---------------------------------------
   Community plugins:
```

**Comment by gnuechtel** [comment]
I guess the lines to change are here:

https://github.com/nrwl/nx/blob/d0ece6e1d2efdf3eb764776ef5271943919956e3/packages/nx/src/generators/utils/project-configuration.ts#L322,L323

Should not be used `joinPathFragments` instead of `join`?

**Comment by AgentEnder** [comment]
Feel free to throw a PR up, I'll assign to you.

**Comment by gnuechtel** [comment]
@AgentEnder PR #10383 is ready to review. Seems to work as expected. It took me some time to setup the local Nx development environment with the local registry, but eventually this issue is fixed by the changes of the PR. The only thing I wonder, is this code already covered by some automated test or would it make sense to create one and how?

**Comment by ShiJuuRoku** [comment]
I have same issue when made `nx g mv` with `Nx 14.1.9`
 - `.eslintrc.json`  (`extends` property)
 -  `ng-package.json` (`dest` property)
 - `project.json` (`$schema` property)

**Comment by gnuechtel** [comment]
Fixed by commit 9e6e01b
